typing/oprint.ml

```diff
@@ -22,11 +22,13 @@ let cautious f ppf arg =
   try f ppf arg with
     Ellipsis -> fprintf ppf "..."
 
+let fix_ident = function "::" -> "(::)" | s -> s
```

Comment:
This approach is correct, but a more elegant approach would be to consider the type `out_ident` as if it was defined by the more informative

```
type out_ident =
  | Oide_apply of out_ident * out_ident
  | Oide_dot of out_ident * out_lident
  | Oide_ident of out_lident
and out_lident = string (* lident: short lowercase identifier *)
```
and have a `print_lident ppf s` function that internally applies the `fix` logic.

(There may be a better name than `lident`; in the parser lident/uident clearly refer to path *components* without dots, but in the typer "ident" rather refers to "long idents" that are in fact value paths. Maybe "name" would work, with the idea that a value path is a (possibly empty) module path followed by a value name, but it is less self-evident.)